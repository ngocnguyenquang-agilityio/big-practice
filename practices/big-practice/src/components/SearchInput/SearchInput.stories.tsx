// Types
import type { Meta, StoryObj } from '@storybook/react';
import type { InputProps } from '@components/Input';

// Components
import SearchInput from '.';

export default {
  title: 'Components/SearchInput',
  component: SearchInput,
  render: (args) => <SearchInput onChange={() => { }} {...args} />
} as Meta

type Story = StoryObj<InputProps>

export const Default: Story = {
  args: {
    type: 'text',
    placeholder: 'Search Input',
    onChange: () => { }
  }
}
