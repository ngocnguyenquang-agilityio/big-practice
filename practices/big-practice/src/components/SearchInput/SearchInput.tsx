// Icons
import { Icon } from '@components/Icon/Icon';
import searchIcon from '@assets/searchIcon.svg';

// Components
import { Input } from '@components/Input/Input';

// Types
import { InputProps } from '@components/Input/Input';

const SearchInput = ({ ...rest }: InputProps): React.ReactElement => {
  return (
    <>
      <Input {...rest} />
      <div className='absolute inset-y-0 right-0 flex items-center pr-4 pointer-events-none'>
        <Icon
          svg={searchIcon}
          name='search-icon'
        />
      </div>
    </>
  );
};

export default SearchInput;
