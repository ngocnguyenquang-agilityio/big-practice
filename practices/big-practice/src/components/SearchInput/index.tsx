// Icons
import { Icon } from '@components/Icon';
import searchIcon from '@assets/searchIcon.svg';

// Components
import { Input } from '@components/Input';

// Types
import { InputProps } from '@components/Input';

const SearchInput = ({ ...rest }: InputProps): React.ReactElement => {
  return (
    <>
      <Input {...rest} />
      <div className='absolute inset-y-0 right-0 flex items-center pr-4 pointer-events-none'>
        <Icon
          svg={searchIcon}
          name='search-icon'
        />
      </div>
    </>
  );
};

export default SearchInput;
