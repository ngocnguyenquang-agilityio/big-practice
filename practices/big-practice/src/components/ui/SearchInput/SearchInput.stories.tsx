import type { Meta, StoryObj } from '@storybook/react';
import SearchInput from './SearchInput';
import { InputProps } from '../Input/input';

export default {
  title: 'Components/SearchInput',
  component: SearchInput,
  render: (args) => <SearchInput onChange={() => { }} {...args} />
} as Meta

type Story = StoryObj<InputProps>

export const Default: Story = {
  args: {
    type: 'text',
    placeholder: 'Search Input',
    onChange: () => { }
  }
}
