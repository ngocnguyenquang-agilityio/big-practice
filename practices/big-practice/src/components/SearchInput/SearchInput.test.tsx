import { render, fireEvent } from '@testing-library/react';

// Types
import type { InputProps } from '@components/Input';

// Components
import SearchInput from '.';

const props: InputProps = {
  type: 'text',
  placeholder: 'Search products ...',
  onChange: jest.fn()
}

describe("SearchInput component", () => {
  test('Should render successfully', () => {
    const comp = render(<SearchInput {...props} />)
    expect(comp).toBeTruthy();
  })

  test('Should match snapshot', () => {
    const comp = render(<SearchInput {...props} />)
    expect(comp).toMatchSnapshot();
  })

  test('Should able to type', () => {
    const { getByPlaceholderText } = render(<SearchInput {...props} />);
    const searchInput = getByPlaceholderText('Search products ...') as HTMLInputElement;
    const value = 'typing search value'

    fireEvent.change(searchInput, { target: { value: value } });
    expect(searchInput.value).toBe(value)
  })
})
