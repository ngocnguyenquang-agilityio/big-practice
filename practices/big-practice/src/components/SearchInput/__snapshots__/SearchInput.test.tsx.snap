// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="w-full rounded-lg border border-neutral-800 bg-transparent px-4 py-2 text-sm text-white placeholder:text-neutral-400"
        placeholder="Search products ..."
        type="text"
      />
      <div
        class="absolute inset-y-0 right-0 flex items-center pr-4 pointer-events-none"
      >
        <img
          alt="search-icon"
          src=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <input
      class="w-full rounded-lg border border-neutral-800 bg-transparent px-4 py-2 text-sm text-white placeholder:text-neutral-400"
      placeholder="Search products ..."
      type="text"
    />
    <div
      class="absolute inset-y-0 right-0 flex items-center pr-4 pointer-events-none"
    >
      <img
        alt="search-icon"
        src=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
